applications/RomApplication/tests/test_structural_lspg_rom.py

```diff
@@ -0,0 +1,122 @@
+            aux_nodal_area = rom_testing_utilities.GetNodalAreaVector(self.simulation._GetSolver().GetComputingModelPart())
+            nodal_area = [aux_nodal_area[int((i-1)/2) if i%2 != 0 else int(i/2)] for i in range(2*len(aux_nodal_area))]
+
+            numerator = 0.0
+            denominator = 0.0
+            for i in range(len(obtained_output)):
+                if expected_output[i] > 1.0e-12:
+                    numerator += nodal_area[i]*((1 - (obtained_output[i]/expected_output[i]))**2)
+                    denominator += nodal_area[i]
+            l2 = np.sqrt(numerator/denominator)*100
+            self.assertLess(l2, self.relative_tolerance)
+
+            # Save the obtained results in TEMPERATURE variable and visualize
+            output_results = False
+            if output_results:
+                i = 0
+                for node in self.simulation._GetSolver().GetComputingModelPart().Nodes:
+                    node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_X, 0, obtained_output[i])
+                    i += 1
+                    node.SetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y, 0, obtained_output[i])
+                    i += 1
+                self.PostProcess("test_rbf_shape_functions_utility_1x1_square_G")
+        
+
+    def testStructuralDynamicLSPGRom2D(self):
+        self.work_folder = "structural_dynamic_test_files/LSPGROM/"
+        parameters_filename = "ProjectParametersLSPGROM.json"
+        expected_output_filename = "ExpectedOutputLSPGROM.npy"
+
+        time_snapshots = [2,4,6,8,10]
+
+        with KratosUnittest.WorkFolderScope(self.work_folder, __file__):
+            # Set up and run simulation
+            with open(parameters_filename,'r') as parameter_file:
+                parameters = KratosMultiphysics.Parameters(parameter_file.read())
+            model = KratosMultiphysics.Model()
+            self.simulation = rom_testing_utilities.SetUpSimulationInstance(model, parameters)
+
+            # Patch the RomAnalysis class to save the selected time steps results
+            def Initialize(cls):
+                super(type(self.simulation), cls).Initialize()
+                cls.selected_time_step_solution_container = []
+
+            def FinalizeSolutionStep(cls):
+                super(type(self.simulation), cls).FinalizeSolutionStep()
+
+                time = cls._GetSolver().GetComputingModelPart().ProcessInfo[KratosMultiphysics.TIME]
+                if np.any(np.isclose(time, time_snapshots)):
+                    array_of_displacements = rom_testing_utilities.GetVectorNodalResults(self.simulation._GetSolver().GetComputingModelPart(), KratosMultiphysics.DISPLACEMENT)
+                    cls.selected_time_step_solution_container.append(array_of_displacements)
+
+            self.simulation.Initialize  = types.MethodType(Initialize, self.simulation)
+            self.simulation.FinalizeSolutionStep  = types.MethodType(FinalizeSolutionStep, self.simulation)
```

Comment:
What is the point on this?

Reply:
to susbtitute the given method inside the class (therefore the python2 style in the super() was required)